src/components/modebar/buttons.js

```diff
@@ -347,15 +347,19 @@ function handleCamera3d(gd, ev) {
         var key = sceneId + '.camera';
         var scene = fullLayout[sceneId]._scene;
 
+        if(attr === 'resetDefault' || attr === 'resetLastSave') {
+            aobj[key] = Lib.extendDeep({}, scene.viewInitial);
+
+            aobj[key].projection = {
```

Comment:
We shouldn't need this after 9e1aa7af86a8dab6e807e8ddea8dca826031cc41

Reply:
Actually, you'll need to tweak `aobj` / `key` logic. In the end, `aobj` should look like:

```js
aobj = {
  'scene.camera.up': {/**/},
  'scene.camera.eye': {/**/},
  'scene.camera.center': {/**/}
}
```

as opposed to

```js
aobj = {
  'scene.camera': {
    up: {/**/},
    eye: {/**},
    center: {/**}
  }
}
```